library/summary.ml

```diff
@@ -89,6 +89,13 @@ let unfreeze_single name state =
       Pp.(seq [str "Error unfreezing summary "; str name; fnl (); CErrors.iprint e]);
     iraise e
 
+let warn_summary_out_of_scope =
+  let name = "summary-out-of-scope" in
+  let category = "pedantic" in
+  let default = CWarnings.Disabled in
+  CWarnings.create ~name ~category ~default (fun name ->
+       Pp.(str "Summary was captured out of module scope for entry " ++ str name))
```

Comment:
I'd suggest a more informative message, such as "A Coq plugin was loaded inside a local scope (such as a Section). It is recommended to load plugins at the start of the file.

Reply:
I was expecting that too, but there is another warning below. Maybe this one is capturing other scenarios?

Reply:
Well for this code I would indeed do just a `msg_debug` flag, as I see it more as a dev-only warning. Once we get the configure profiles in place it should be easy to tweak this stuff better.

Reply:
What I suggest is to keep using a warning, but have a dedicated "dev" category for them. Those warnings should be off by default, and turned on for the developer profile.

Reply:
> "dev"

(feel free to pick another name)